### PR TITLE
refactor: reduce memory usage

### PR DESCRIPTION
**Pull request by flavio**
Re-architect how the waPC policies are handled. This is required to reduce the amount of memory used by Policy Server.

I'm sorry this is a pretty big PR, but there was no way to split it into smaller units :(

### CODE REVIEW

**Comment by flavio** [comment]
I've rebased against the latest changes on the `main` branch

File: src/policy_evaluator.rs [hunk 1]
```diff
@@ -7,8 +7,10 @@ use std::{convert::TryFrom, fmt};
 
 use crate::admission_request::AdmissionRequest;
 use crate::admission_response::AdmissionResponse;
-use crate::policy::Policy;
+use crate::evaluation_context::EvaluationContext;
+// use crate::policy::Policy;
```
Comment (fabriziosestito) [inline src/policy_evaluator.rs:11]: leftover

Comment (flavio) [reply src/policy_evaluator.rs]: thanks, fixed!

**Review by fabriziosestito — APPROVED** [review]
LGTM, brilliant!

File: src/runtimes/wapc/evaluation_context_registry.rs [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+};
+use tracing::debug;
+
+use crate::evaluation_context::EvaluationContext;
+
+lazy_static! {
+    /// A map with wapc_id as key, and the worker_id as value. It allows us to know to which
+    /// worker a waPC policy belongs. When inside of the
+    /// [`host_callback`](crate::runtimes::wapc::callback::host_callback) function,
+    /// we need to know
```
Comment (viccuad) [inline src/runtimes/wapc/evaluation_context_registry.rs:65]: this last phrase sounds cut. "need to know the EvaluationContext" perhaps?

**Review by viccuad — APPROVED** [review]
LGTM! Great!

File: src/runtimes/wapc/evaluation_context_registry.rs (hunk 2 above)
Comment (flavio) [reply src/runtimes/wapc/evaluation_context_registry.rs]: thanks, I've fixed it

**Pull request merged**